src/native/eventpipe/ep-session.c

```diff
@@ -34,6 +34,19 @@ ep_session_remove_dangling_session_states (EventPipeSession *session);
  * EventPipeSession.
  */
 
+static size_t streaming_loop_tick(EventPipeSession *const session, bool *events_written) {
+	if (!ep_session_get_streaming_enabled (session)){
+		session->streaming_thread = NULL;
+		return 1; // done
+	}
+	if (!ep_session_write_all_buffers_to_file (session, events_written)) {
+		session->streaming_thread = NULL;
+		ep_disable ((EventPipeSessionID)session);
```

Comment:
I think we need to preserve this call to happen outside the EP_GCX_PREEMP_ENTER/EXIT as it did before.

Reply:
Could you please explain why ?

Reply:
ep_disable can run managed code so thread needs to be back in coop mode before running that call.